### ISSUE TITLE
[Circuit-UI] Textarea component - new functionality (expand the component by the content)

### ISSUE THREAD

**Issue by Marczerwinski** [sumup-oss/circuit-ui#1093]
## Component to amend

[TEXTAREA](https://circuit.sumup.com/?path=/docs/forms-textarea--base)

## Visual

![image](https://user-images.githubusercontent.com/18164156/128852942-3b3bfae4-981a-4992-b5bd-5117dbd846db.png)

## Context

Suggested by design team functionality update [here](https://sumupteam.atlassian.net/browse/SA-47789).

```
A maximum of two lines of text is visible without interaction. To view the rest of the text the user must either: 

scroll the text field, or

Manually expand it themselves using the grips in the bottom right hand corner of the control

This is an uncommon pattern and its affordance is overridden by the presence of a scroll bar with stronger affordance in the textfield, therefore a majority of our users will not know/find this manual expand functionality.

Manual expansion does not help when one is in the process of entering copy into the textfield, however automatic expansion of the field does.
```

**Comment by robinmetral** [comment]
Hi!

That seems like a good idea–we will talk with Giao-Chung about this to see if it should become the default behavior, the most important being to keep our experience consistent (so thank you for bringing this to Circuit!).

In the meantime, how would you suggest implementing this? Will you be able to contribute it?

There are existing examples in [MUI](https://github.com/mui-org/material-ui/blob/f336e03acfadfe0171cee5a469be02953337d20f/packages/material-ui/src/TextareaAutosize/TextareaAutosize.js) or in [third-party libraries](https://github.com/Andarist/react-textarea-autosize), we can probably do something similar (but lighter?) in Circuit.

Looking forward to hearing your thoughts!

**Comment by Marczerwinski** [comment]
@robinmetral yes, we can check how it's made with existing examples and implement something similar. We can also go with some solutions based on scroll height (also possible solution). I would really like to take part in implementing this feature (if it's gonna be implemented). I would like to see (or take part) how the process of implementing changes in circuit looks from the 'inside' 🙂

**Comment by robinmetral** [comment]
This would be amazing! Then if you'd like to give this a shot, I would suggest  starting by going through the [contribution guidelines](https://circuit.sumup.com/?path=/story/introduction-contributing-overview--page). It has docs on how to get your local environment up and running and best practices for our code conventions and contribution etiquette.

I'll also let you know if this can become the new default or if this should only be an option after I talk with design.

Feel free to reach out if I can help in any way!
